core/src/main/java/org/mule/routing/MessageFilter.java

```diff
@@ -90,7 +90,7 @@ protected boolean accept(MuleEvent event)
 
         if (event != null && !VoidMuleEvent.getInstance().equals(event))
```

Comment:
Don't disagree but don't see how this is related to this PR, especially as currently Filter interface still supports Message and Event and it is accept(Message msg) that implementers must implement still.